### PR TITLE
add error component to app for when the apis are broken

### PR DESCRIPTION
**Pull request by kendallm360**
## Summary:
Add error component to the app in the event the apis break

## Type of Changes:
- [ ] Bug fix
- [x] Refactor
- [x] New feature

## Where are the Changes:
App.tsx
AmarettoDeck.tsx
MartiniDeck.tsx

## How was this tested:
Chrome/React dev tools

### CODE REVIEW

**Review by nhodnett — APPROVED** [review]
Nice work on the error handling, and the refactor of some of the css!

File: src/components/AmarettoDeck/index.tsx [hunk 1]
```diff
@@ -14,28 +15,36 @@ class AmarettoDeck extends Component<{}, TypeState> {
...
   componentDidMount = () => {
-    fetchSpecialtyCocktails("search.php?s=amaretto").then((data) => {
-      this.setState({ drinkList: data.drinks });
-    });
+    fetchSpecialtyCocktails("search.php?s=amaretto")
+      .then((data) => {
+        this.setState({ drinkList: data.drinks });
+      })
+      .catch(() => {
+        this.setState({ error: true });
```
Comment (jordan-sullivan) [inline src/components/AmarettoDeck/index.tsx:23]: nice job adding error boolean to state.

File: src/apiCalls.tsx [hunk 2]
```diff
@@ -2,19 +2,19 @@ const baseURL = "https://www.thecocktaildb.com/api/json/v1/1/";
 
 const fetchAllCocktails = () => {
   return fetch(`${baseURL}filter.php?g=Cocktail_glass`).then((response) => {
-    if(response.ok) {
-      return response.json()
+    if (response.ok) {
```
Comment (jordan-sullivan) [inline src/apiCalls.tsx:5]: nice way to check response, glad we could implement!

File: src/components/AmarettoDeck/index.tsx [hunk 3]
```diff
@@ -14,28 +15,36 @@ class AmarettoDeck extends Component<{}, TypeState> {
...
+        </div>
       );
     });
   };
 
   render() {
-    return <div className="AmarettoContainer">{this.formatAmarettos()}</div>;
+    return (
+      <div className="AmarettoContainer">
+        {this.state.error ? <Error /> : <>{this.formatAmarettos()}</>}
```
Comment (jordan-sullivan) [inline src/components/AmarettoDeck/index.tsx:45]: great logic here! -J

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -28,30 +28,45 @@ class App extends React.Component<{}, TypeState> {
...
+            /> */}
+            <AppContainer
+              drinkList={this.state.drinkList}
+              error={this.state.error}
+            />
           </div>
         </Route>
-        
+
+        <Route exact path="/?/error">
```
Comment (jordan-sullivan) [inline src/components/App/App.tsx:67]: great use of Route

File: src/components/Error/Error.css [hunk 5]
```diff
@@ -1,4 +1,17 @@
...
\ No newline at end of file
+  font-size: 50px;
+  color: red;
+}
+
+.errorImage {
+  width: 30%;
+}
+
+.errorDiv {
```
Comment (jordan-sullivan) [inline src/components/Error/Error.css:10]: nice job ob error styling. Especially love the image. -Jordan

File: src/components/Types.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 export type TypeState = {
   drinkList: Drink[];
   id?: string;
-  error: false | true
```
Comment (jordan-sullivan) [inline src/components/Types.ts:4]: nice use of specific values in TS

**Review by jordan-sullivan — APPROVED** [review]

**Pull request merged**